### PR TITLE
Adds ShaderX Python Wrappers.

### PR DESCRIPTION
**Pull request by feldstj**
Currently all of the files are a part of a single library. After speaking with @bernardkwok , I'll look into splitting this up into multiple libs in the next review.

Also, I currently have a single test which is a port from ShaderGen, but will look into adding a ported version of the main ShaderValid test soon.

### CODE REVIEW

File: python/MaterialXTest/shaderx.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import math
+import os
+import unittest
+
+import MaterialX as mx
+
+_fileDir = os.path.dirname(os.path.abspath(__file__))
+
+def _getSubDirectories(libraryPath):
```
Comment (bernardkwok) [inline python/MaterialXTest/shaderx.py]: I guess this was due to the fact there is no FilePath Pythong wrappers. Maybe ask about this to find out if this was an oversight or on purpose.  As mentioned I don't think we want loadLibraries (plural) is always the correct logic, but maybe  getSubDirs, getMtxFiles and loadLibrary may be.

File: source/PyMaterialX/PyFile.cpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include <MaterialXFormat/File.h>
+
+namespace py = pybind11;
+namespace mx = MaterialX;
+
+void bindPyFile(py::module& mod)
+{
+    py::enum_<mx::FilePath::Type>(mod, "Type")
+        .value("TypeRelative", mx::FilePath::Type::TypeRelative)
+        .value("TypeRelative", mx::FilePath::Type::TypeAbsolute)
```
Comment (bernardkwok) [inline source/PyMaterialX/PyFile.cpp]: Cut and Paste typos here.

File: source/PyMaterialX/PyGenOptions.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+void bindPyGenOptions(py::module& mod)
+{
+    py::enum_<mx::ShaderInterfaceType>(mod, "ShaderInterfaceType")
+        .value("SHADER_INTERFACE_COMPLETE", mx::ShaderInterfaceType::SHADER_INTERFACE_COMPLETE)
+        .value("SHADER_INTERFACE_REDUCED", mx::ShaderInterfaceType::SHADER_INTERFACE_REDUCED);
+
+    py::class_<mx::GenOptions>(mod, "GenOptions")
+        .def_readwrite("shaderInterfaceType", &mx::GenOptions::shaderInterfaceType)
+        .def_readwrite("hwTransparency", &mx::GenOptions::hwTransparency)
```
Comment (bernardkwok) [inline source/PyMaterialX/PyGenOptions.cpp]: I'm wondering if we should fix this name in C++ and here. hw is kind-of meaningless unless your familiar with it. Maybe use the full string "hardware" ?

File: source/PyMaterialX/PyShader.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    py::class_<mx::Shader::VariableBlock, mx::Shader::VariableBlockPtr>(mod, "VariableBlock")
+        .def(py::init([](const std::string& n, const std::string& i) { return new mx::Shader::VariableBlock(n, i); }))
+        .def("empty", &mx::Shader::VariableBlock::empty)
+        .def("size", &mx::Shader::VariableBlock::size)
+        .def("__len__", &mx::Shader::VariableBlock::size)
+        .def("__getitem__", [](const mx::Shader::VariableBlock &vb, size_t i)
+        {
+            if (i >= vb.size()) throw py::index_error();
+            return vb[i];
+        }, py::return_value_policy::reference_internal)
```
Comment (bernardkwok) [inline source/PyMaterialX/PyShader.cpp]: Guess this ended up working after-all :)

Comment (feldstj) [reply source/PyMaterialX/PyShader.cpp]: Hehe, seems to work, was a tricky issue to figure out, thanks for the help!

File: source/PyMaterialX/PyGenOptions.cpp (hunk 3 above)
Comment (feldstj) [reply source/PyMaterialX/PyGenOptions.cpp]: That's fine by me, I'll make the change.

Comment (feldstj) [reply source/PyMaterialX/PyGenOptions.cpp]: Actually as discussed, we use hw in a bunch of places, so leaving this alone for now.

File: python/MaterialXTest/shaderx.py (hunk 1 above)
Comment (feldstj) [reply python/MaterialXTest/shaderx.py]: As discussed will look into this in a future review.

**Pull request merged**